flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/common/CommonExecLookupJoin.java

```diff
@@ -411,6 +426,55 @@ private Transformation<RowData> createSyncLookupJoinWithState(
         return transform;
     }
 
+    /**
+     * Update description of the transformation for easier debugging during job running. Will do:
+     *
+     * <p>1. correct async lookup mode if fallback to sync lookup
+     *
+     * <p>2. append async lookup parameters
+     *
+     * <p>3. append retry strategy
+     */
+    private String updateDescription(
```

Comment:
could we make this class immutable ? Many field values(such as: isAsyncEnabled, asyncLookupOptions) can be determined when translating StreamPhysicalLookupJoin to StreamExecLookupJoin

Reply:
After confirmed that we can determine if async lookup will be enabled for runtime by avoid creating lookup function instance, we should do the change to make exec lookup join node immutable.